shared/js/initApp.js

```diff
@@ -118,9 +119,42 @@ window.apps = {
       }
     })(appOptions.level);
   },
+
+  // Set up projects, skipping blockly-specific steps. Designed for use
+  // by levels of type "external".
+  setupProjectsExternal: function() {
+    dashboard.project = project;
```

Comment:
I've added those lines here. I haven't replaced those lines with a call to this function, because the rest of setup/setupBlockly depends on window.dashboard and it seems like it shouldn't depend on projects setup code to enforce that.